3.analyze_data/UMAP_analysis/nbconverted/2.jump_UMAP.py

```diff
@@ -0,0 +1,389 @@
+filtered_prob_df["Metadata_Predicted_Class"] = filtered_prob_df.iloc[:, :-7].idxmax(
+    axis=1
+)
+
+# Include a new column called Metadata_Phenotypic_Value as to be able to plot on UMAP
+# Take the highest prob value and add as a value in the column to use for labelling
+filtered_prob_df["Metadata_Phenotypic_Value"] = filtered_prob_df.iloc[:, :-7].max(
+    axis=1
+)
+
+print(filtered_prob_df.shape)
+filtered_prob_df.head()
+
+
+# In[6]:
+
+
+# Specify the columns you want to use for merging morphology and probabilities
+merge_columns = [
+    "Metadata_Plate",
+    "Metadata_Well",
+    "Metadata_Site",
+    "Metadata_ObjectNumber_cytoplasm",
+]
+
+# Select only the columns in filtered_prob_df that start with "Metadata"
+filtered_prob_df_subset = filtered_prob_df.filter(like="Metadata")
+
+# Merge the data frames on the specified columns
+merged_prob_df = filter_df.merge(filtered_prob_df_subset, on=merge_columns, how="inner")
+
+# Remove rows with NaN in feature columns
+merged_prob_df = merged_prob_df.dropna(
+    subset=(col for col in merged_prob_df.columns if not col.startswith("Metadata"))
+)
+
+# reset index
+merged_prob_df.reset_index(inplace=True, drop=True)
+
+# Add data frame to dictionary
+df_dict["Only_JUMP_all_features"] = merged_prob_df
+
+print(merged_prob_df.shape)
+merged_prob_df.head(2)
+
+
+# ## Combine data frames
+
+# ### Remove all metadata and only include a metadata for data type
+
+# In[7]:
+
+
+# Find all columns that start with Metadata
+metadata_cols = [col for col in merged_prob_df.columns if col.startswith("Metadata")]
+
+# Create a new DataFrame by selecting only "Metadata_Predicted_Class" and all other columns
+jump_df = merged_prob_df[
+    ["Metadata_Predicted_Class", "Metadata_Phenotypic_Value"]
+    + [col for col in merged_prob_df.columns if col not in metadata_cols]
+]
+
+# Add a data name column to separate between datasets
+jump_df.insert(0, "Metadata_data_name", "jump")
+
+print(jump_df.shape)
+jump_df.head(2)
+
+
+# ### Load in Mitocheck labeled data and update CellProfiler columns to match naming for JUMP
+
+# In[8]:
+
+
+# Load in labeled mitocheck data
+label_df = pd.read_csv(label_data_path)
+
+# Extract feature columns from the JUMP data filtered by phenotypic profiling model features
+feature_cols = [col for col in label_df.columns if col.startswith("CP__")]
+metadata_cols = [col for col in label_df.columns if col.startswith("Metadata_")]
+
+# Filter df with only CP features and the metadata columns
+mito_cp_df = label_df[
+    ["Mitocheck_Phenotypic_Class"]
+    + ["Cell_UUID"]
+    + ["Location_Center_X"]
+    + ["Location_Center_Y"]
+    + metadata_cols
+    + feature_cols
+]
+
+# change prefix for columns to match JUMP
+mito_cp_df.columns = mito_cp_df.columns.str.replace("CP__", "Nuclei_")
+
+print(mito_cp_df.shape)
+mito_cp_df.head()
+
+
+# ### Filter the mitocheck data
+
+# In[9]:
+
+
+# Load in labeled mitocheck data
+label_df = pd.read_csv(label_data_path)
```

Comment:
Looks like you're reading this file in twice?

Reply:
(line 187)

Reply:
Weird, great catch! I guess I do. I will remove the later once since it was loaded in earlier. Thanks!